internal/metamorph/callbacker.go

```diff
@@ -30,15 +31,17 @@ type Callback struct {
 }
 
 type Callbacker struct {
-	httpClient HttpClient
-	wg         sync.WaitGroup
-	mu         sync.Mutex
-	disposed   bool
+	httpClient     HttpClient
+	wg             sync.WaitGroup
+	mu             sync.Mutex
+	disposed       bool
+	processorStats *ProcessorStatsCollector
 }
 
-func NewCallbacker(httpClient HttpClient) *Callbacker {
+func NewCallbacker(httpClient HttpClient, processorStats *ProcessorStatsCollector) *Callbacker {
 	return &Callbacker{
-		httpClient: httpClient,
+		httpClient:     httpClient,
+		processorStats: processorStats,
 	}
 }
```

Comment:
I think these metrics could be added in `callbacker.go` to another struct like `callbackerStats`. Then there is no need to change `stats_collector.go` and making that type public

They can be registered in the constructor with `prometheus.Register()` and unregistered in `Shutdown()`

```
	CallbackSeenOnNetworkCount       prometheus.Gauge
	CallbackSeenInOrphanMempoolCount prometheus.Gauge
	CallbackRejectedCount            prometheus.Gauge
	CallbackMinedCount               prometheus.Gauge
	CallbackFailedCount              prometheus.Gauge
```

Also the type can be `prometheus.Counter`, as the value can only go up and never down.